.github/workflows/e2e-kind.yaml

```diff
@@ -0,0 +1,144 @@
+name: E2E tests on kind
+
+on:
+  issue_comment:
+    types: [created]
+
+permissions:
+  contents: read
+  pull-requests: write
+  checks: write
+
+jobs:
+  check-comment:
+    name: Check for trigger comment
+    if: github.event.issue.pull_request && contains(github.event.comment.body, '/run-e2e-kind')
+    runs-on: ubuntu-24.04
+    outputs:
+      triggered: ${{ steps.check.outputs.triggered }}
+      pr_ref: ${{ steps.get-pr.outputs.ref }}
+      pr_sha: ${{ steps.get-pr.outputs.sha }}
+    steps:
+      - name: Check comment author
+        id: check
+        run: |
+          echo "triggered=true" >> $GITHUB_OUTPUT
+      - name: Get PR details
+        id: get-pr
+        uses: actions/github-script@v7
+        with:
+          script: |
+            // When a workflow is triggered by an issue_comment event the ref and sha vars refer
+            // to the base branch and commit of the workflow run, not the pull request's head branch or commit.
+            // Hence we need to explicitly fetch them from the PR details.
+
+            const pr = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            });
+            core.setOutput('ref', pr.data.head.ref);
+            core.setOutput('sha', pr.data.head.sha);
+
+  run-e2e-kind:
+    name: Run E2E tests on kind
+    needs: check-comment
+    if: needs.check-comment.outputs.triggered == 'true'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Comment with workflow link
+        uses: actions/github-script@v7
+        with:
+          script: |
+            await github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+              body: `🚀 E2E workflow started! [View workflow run](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})\n\nℹ️ Please note that currently only the edgeconnect tests without a proxy (\`test/e2e/edgeconnect/normal\`) are being run.`
+            });
+      - name: Checkout PR
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          ref: ${{ needs.check-comment.outputs.pr_sha }}
+          fetch-depth: 0
+      - name: Setup up kind
+        uses: helm/kind-action@92086f6be054225fa813e0a4b13787fc9088faab # v1.13.0
+        with:
+          # renovate github-releases depName=github.com/kubernetes-sigs/kind
+          version: v0.30.0
+          install_only: true
+      - name: Set up go
+        uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00 # v6.0.0
+        with:
+          go-version-file: "./go.mod"
+      - name: Set up helm
+        uses: azure/setup-helm@1a275c3b69536ee54be43f2070a358922e12c8d4 # v4.3.1
+        with:
+          # usually we use latest, but 3.18.0 has bug https://github.com/helm/helm/issues/30890
```

Comment:
How do we want keep track of that? If there comes out a new version that is bugfree should we just manually update?